### ISSUE TITLE
Merge readthedocs French and English sites

### ISSUE THREAD

**Issue by DeflateAwning** [braillerap/BrailleRap#6]
I'm having trouble figuring out how to contribute to the English BrailleRAP readthedocs. They seem to be stored somewhat as a translation, but are hosted on their own sites (i.e., at `https://braillerap-en.readthedocs.io/` instead of `https://braillerap.readthedocs.io/en`.

Would appreciate more info about these, and think that maybe the two sites should be merged.

**Comment by crocsg** [comment]
There is a little bit of history in that. When we started the documentation on readthedoc there was no very clear way to make multilingual documentation project. Since readthedoc use python Sphynx, we translated the french documentation with Sphynx and build a separate project for the english version. Beside the 2 url i have many concern about this process.
- The documentation structure is defined in French. For years it was not a problem as most of BrailleRAP builder where French locutor. Things have changed in the past few month.
- To get effective translation in a language, you need to build a Sphynx pipeline somewhere. This is not a problem for Python user, but it is obviously a bottleneck in new translation contribution.
- As the project evolve, it's becoming more and more difficult to well document some of the device options. I made something with BrailleRAP and BrailleRAP XL, but i have some experiment working with different kind of controller board and would like to publish it without disturbing new BrailleRAP builders. There is also some experiment around the worlds (indonesian guys build some with a complete redesign of the left and right frame just printed on bambulab 3d printer, the guys are not very aware of git and i'm still working to have all that experiment published and "linked" to the BrailleRAP project.
- The actual documentation structure as been refined in public workshop to build the device. It's quite nice for workshop animation, but maybe not the best one for a new BrailleRAP enthusiast.

So i agree, we need to change the documentation, something more modular, more user friendly and make it easier to have new languages. I had a great experiment with weblate on codeberg and BrailleRAP softwares. It's a little different to translate full text than software label, but i'm curious to give it a try. It seem that there is some work to do, maybe not as much as i think, but at this time i'm focused on BrailleRAP softwares (and there will be a software documentation subject at the corner). Maybe it would be a good idea to start a new documentation project, starting with resources from the actual one, and change the background process.  As soon as we are ok with the new process, it's easy to make it the "official" documentation of the project


And just for technical precision, the documentation in french is located in .rst at ./doc the en translation is located at ./doc/locale/en/LC_MESSAGES the .pot needed for gettext are at ./doc/_build/gettext

**Comment by DeflateAwning** [comment]
It seems like the `_build/` folder should be in the gitignore. It's not really standard to version control build products.

I'd recommend picking a documentation system that's a bit simpler. Maybe just having the documentation is Markdown files, so they can be viewed directly with git? And then the translations are just completely rewriting each file in a given locale?

**Comment by crocsg** [comment]
It seem a bunch of file in git repo will not serve the purpose. Has i've already said, 80% of the BrailleRAP community just don't know what git is

**Comment by crocsg** [comment]
Close this until some time and resources to find a better alternative than readthedoc

**Comment by DeflateAwning** [comment]
Where's this community you speak of? You didn't even have Issues turned on, and there's no other communication channel on any resources. 

No one know how git/github work when they're born. Using them is the way you learn. I'm not sure why you're so afraid of people seeing git being used well, and then learning how to use it themselves. Strange.

I was suggesting that you _store_ the docs in Git and serve them on a website, if you're trying to avoid git.

I suggest reopening the issue as a place to centralize discussion about this; alternatively, you can make a new Issue and move the discussion there. 

GitHub Pages is probably your best bet, if you want to move off of readthedocs. GitHub Pages serves the documentation like a regular website, and is built from CI actions in GitHub (which I'd be happy to help setup, if you'd like). Or did you have other ideas?

**Comment by crocsg** [comment]
Maybe some life exists outside of github, who knows ?

> I suggest reopening the issue as a place to centralize discussion about this; alternatively, you can make a new Issue and move the discussion there.

Good idea, i've just one old brain, and feel uncomfortable to handle the same discussion over three different channels.

What i suggest is to discuss about the goals a little before jumping on the first pretty technology. Has everybody, i guess you are not happy to work on something, and put it in the bin because you miss the goal.

 If you try to get some information about this projects, you will find that
- Some of our great contributors are unsighted, github is not truly accessible, and you can't really use it only with the command line. But they can do something that i guess you can't, read Braille with their fingers, have some valuable opinion about the quality of documents produced by the device. That's one of the paradox of BrailleRAP, we are designing devices for unsighted peoples, and we just absolutely need them to qualify the device.
- Some of our contributors are far away from IT techs, because they are teachers, sociologists or NGO volunteers. they don't give a shit about git or github, just because they fight everyday to get unsighted children at school learning some Braille. Can you answer the following questions ? Is a BrailleRAP suitable for  usage in school ? and in a Museum ? How is it difficult to build one in Europe or India ? and in the middle of Africa , in Brasil or in Philipina ? Where i can find the 3d model to repair a BrailleRAP built 2 years ago ?
- Some of them "just" contribute with translations via weblate. You can write in english, what about Italian, Helenic, Arabic or Chinese ? Is there some interests to have a building manual in Nepalese or Romanian ? in pdf ? on a mobile ?

In conclusion, take a rest, take some time to think about all that. Any technical contribution would be helpful, but only if they are serving the right goals. And if this project is not fancy enough to put all the tech you are dreaming to use, this is not a problem, this is just the life.  Github is full of projects who need great expertise you can contribute on.
